### ISSUE TITLE
GPU block optimizer state not checkpointed — continuation training broken (ALB-118)

### ISSUE THREAD

**Issue by noahgift** [paiml/entrenar#287]
## Problem

APR checkpoints save CPU embedding optimizer state (`__training__.embed_optimizer.m/v.*`) but NOT GPU-resident `CudaGradWorkspace` AdamW state for 24 transformer blocks (99%+ of parameters).

This makes continuation training impossible:
- **v10**: fresh optimizer + lower LR (1e-4) → val_ppl plateaued at 660
- **v11**: fresh optimizer + same LR (3e-4) with re-warming → val_ppl plateaued at 750
- **v12**: resume path with embed optimizer restored → val_ppl=5639 (one full-LR step destroyed weights)

All three confirmed: Adam second moments encode per-parameter loss landscape curvature. Without GPU block optimizer state, fresh moments produce wrong adaptive step sizes that corrupt the model's learned weights.

## Root Cause

`prepare_async_apr_save()` (cuda_trainer.rs ~line 2688) only snapshots the CPU embedding optimizer:
```rust
// Only embed_optimizer is saved
for (i, chunk) in self.embed_optimizer.first_moments().iter().enumerate() {
    tensors.insert(format!("__training__.embed_optimizer.m.{i}"), ...);
}
```

The GPU block optimizer (24 blocks × ~62 MB m/v each) is never transferred D2H for checkpointing.

## Proposed Fix

1. Add `download_optimizer_state()` to `CudaTransformerBlock` — D2H transfer of AdamW m/v buffers
2. In `prepare_async_apr_save()`, iterate blocks and save `__training__.block.{i}.m/v.*` tensors
3. In `load_optimizer_state_apr()`, restore block m/v from APR
4. Total checkpoint overhead: ~1.5 GB additional per save (24 blocks × 62 MB)
5. Async save already runs in background thread — D2H transfers would add ~50ms per checkpoint

## Acceptance Criterion

Resume from checkpoint produces val_ppl within ±5% of pre-save val_ppl on same validation data.

## References

- albor gap register: ALB-118 in `docs/book/src/spec/11-gaps.md`
- Ibrahim et al. 2024: "Simple and Scalable Strategies to Continually Pre-train LLMs"
- Gupta et al. 2023: "How to (re)warm your model?"
